Api/csharp/ArmoniK.Api.Worker/Worker/WorkerStreamWrapper.cs

```diff
@@ -0,0 +1,103 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+
+using ArmoniK.Api.gRPC.V1;
+using ArmoniK.Api.gRPC.V1.Agent;
+using ArmoniK.Api.gRPC.V1.Worker;
+using ArmoniK.Api.Worker.Utils;
+
+using Grpc.Core;
+
+using JetBrains.Annotations;
+
+using Microsoft.Extensions.Logging;
+
+namespace ArmoniK.Api.Worker.Worker;
+
+[PublicAPI]
+public class WorkerStreamWrapper : Api.gRPC.V1.Worker.Worker.WorkerBase, IAsyncDisposable
+{
+  private readonly ILoggerFactory               loggerFactory_;
+  public           ILogger<WorkerStreamWrapper> logger_;
+  private readonly ChannelBase                  channel_;
+  private readonly Agent.AgentClient            client_;
+
+  public WorkerStreamWrapper(ILoggerFactory      loggerFactory,
+                             GrpcChannelProvider provider)
+  {
+    logger_        = loggerFactory.CreateLogger<WorkerStreamWrapper>();
+    loggerFactory_ = loggerFactory;
+
+
+    logger_.LogDebug("Trying to create channel for {address}",
+                     "/cache/armonik_agent.sock");
+
+    channel_ = provider.Get();
+
+    client_ = new Agent.AgentClient(channel_);
+  }
+
+  public sealed override async Task<ProcessReply> Process(IAsyncStreamReader<ProcessRequest> requestStream,
+                                                          ServerCallContext                  context)
+  {
+    Output output;
+    {
+      await using var taskHandler = await TaskHandler.Create(requestStream,
+                                                             client_,
+                                                             loggerFactory_,
+                                                             context.CancellationToken)
+                                                     .ConfigureAwait(false);
+
+      using var _ = logger_.BeginNamedScope("Execute task",
+                                            ("taskId", taskHandler.TaskId),
+                                            ("sessionId", taskHandler.SessionId));
+      logger_.LogDebug("Execute Process");
+      output = await Process(taskHandler)
+                 .ConfigureAwait(false);
+    }
+    return new ProcessReply
+           {
+             Output = output,
+           };
+  }
+
+  public virtual Task<Output> Process(ITaskHandler taskHandler)
+    => throw new RpcException(new Status(StatusCode.Unimplemented,
+                                         ""));
+
+  public override Task<HealthCheckReply> HealthCheck(Empty             request,
+                                                     ServerCallContext context)
+    => Task.FromResult(new HealthCheckReply
+                       {
+                         Status = HealthCheckReply.Types.ServingStatus.Serving,
+                       });
+
+  public async ValueTask DisposeAsync()
```

Comment:
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#implement-the-async-dispose-pattern

Reply:
adding
`GC.SuppressFinalize(this);` ?